### PR TITLE
updates Twitter and Facebook meta data in head.html.

### PR DESCRIPTION
**Pull request by derekshirk**
## Overview

Tries to fix and pick up where [PR #114](https://github.com/cloudfour/pwastats/pull/114) left off. I still could not get the twitter validator to work without the `twitter:title` meta tag, so I removed the `og:title` tag for now.

---

### CODE REVIEW

File: _includes/head.html [hunk 1]
```diff
@@ -15,5 +15,33 @@
...
+  {% endif %}
+
+  <meta property="og:description" content="{% if page.excerpt %}{{ page.excerpt | strip_html | strip_newlines | truncate: 160 }}{% else %}{{ site.description }}{% endif %}"/>
+  <meta property="og:type" content="website" />
+  <meta property="og:url" content="{{ site.url }}" />
+  <meta property="og:image" content="http://localhost:4000/images/pwa-stats-1280-cropped.png" />
+
+
+
+
```
Comment (tylersticka) [inline _includes/head.html]: Nit: Why all these extra line breaks?

File: _includes/head.html [hunk 2]
```diff
@@ -15,5 +15,33 @@
...
+  {% if page.layout == 'permalink' %}
+  <meta name="twitter:card" content="summary">
+  {% else %}
+  <meta name="twitter:card" content="summary_large_image">
+  {% endif %}
+
+  {% if site.title %}
+  <meta name="twitter:title" content="{% if page.title %}{{ page.title }} - {% endif %}{{ site.title }}">
+  <meta name="twitter:site" content="@pwastats">
+  <meta name="twitter:creator" content="@cloudfour">
```
Comment (tylersticka) [inline _includes/head.html]: Nit: Inconsistent indentation within `{% if %}` blocks. No indentation here, indentation elsewhere.

File: _includes/head.html [hunk 3]
```diff
@@ -15,5 +15,33 @@
...
+  {% endif %}
+
+  {% if page.title %}
+    <meta property="og:site_name" content="{{ site.title }}"/>
+  {% endif %}
+
+  <meta property="og:description" content="{% if page.excerpt %}{{ page.excerpt | strip_html | strip_newlines | truncate: 160 }}{% else %}{{ site.description }}{% endif %}"/>
+  <meta property="og:type" content="website" />
+  <meta property="og:url" content="{{ site.url }}" />
```
Comment (tylersticka) [inline _includes/head.html]: Is this field necessary?

File: _includes/head.html [hunk 4]
```diff
@@ -15,5 +15,33 @@
...
+  <meta name="twitter:site" content="@pwastats">
+  <meta name="twitter:creator" content="@cloudfour">
+  {% endif %}
+
+  {% if page.title %}
+    <meta property="og:site_name" content="{{ site.title }}"/>
+  {% endif %}
+
+  <meta property="og:description" content="{% if page.excerpt %}{{ page.excerpt | strip_html | strip_newlines | truncate: 160 }}{% else %}{{ site.description }}{% endif %}"/>
+  <meta property="og:type" content="website" />
```
Comment (tylersticka) [inline _includes/head.html]: Is this field necessary?

File: _includes/head.html [hunk 5]
```diff
@@ -15,5 +15,30 @@
...
+  {% endif %}
+
+  {% if site.title %}
+  <meta name="twitter:title" content="{% if page.title %}{{ page.title }} - {% endif %}{{ site.title }}">
+  <meta name="twitter:site" content="@pwastats">
+  <meta name="twitter:creator" content="@cloudfour">
+  {% endif %}
+
+  {% if site.title %}
+    <meta property="og:site_name" content="{{ site.title }}"/>
```
Comment (tylersticka) [inline _includes/head.html]: Nit: Inconsistent self-closing tags. Closing `/` is not necessary.

**Comment by derekshirk** [comment]
@tylersticka Updated based on comments.

File: _includes/head.html [hunk 6]
```diff
@@ -15,5 +15,28 @@
...
+  {% endif %}
+
+  {% if page.layout == 'permalink' %}
+    <meta name="twitter:card" content="summary">
+  {% else %}
+    <meta name="twitter:card" content="summary_large_image">
+  {% endif %}
+
+  {% if site.title %}
+    <meta name="twitter:title" content="{% if page.title %}{{ page.title }} - {% endif %}{{ site.title }}">
```
Comment (tylersticka) [inline _includes/head.html]: It seems `og:title` was omitted entirely?

According to Twitter's docs, you should _only_ need `og:title`, but if you're seeing different results, you can combine the two like so:

```
<meta property="og:title" name="twitter:title" content="{% if page.title %}{{ page.title }} - {% endif %}{{ site.title }}">
```

File: _includes/head.html [hunk 7]
```diff
@@ -15,5 +15,28 @@
...
+    <meta name="twitter:site" content="@pwastats">
+    <meta name="twitter:creator" content="@cloudfour">
+  {% endif %}
+
+  {% if site.title %}
+    <meta property="og:site_name" content="{{ site.title }}">
```
Comment (tylersticka) [inline _includes/head.html]: Could you combine these two conditional blocks?

**Comment by derekshirk** [comment]
@tylersticka updated based on your previous comments.

File: _includes/head.html [hunk 8]
```diff
@@ -15,5 +15,24 @@
...
+  {% if page.layout == 'permalink' %}
+    <meta name="twitter:card" content="summary">
+  {% else %}
+    <meta name="twitter:card" content="summary_large_image">
+  {% endif %}
+
+  <meta property="og:description" content="{% if page.excerpt %}{{ page.excerpt | strip_html | strip_newlines | truncate: 160 }}{% else %}{{ site.description }}{% endif %}">
+  <meta property="og:type" content="website" >
+  <meta property="og:url" content="{{ site.url }}" >
+  <meta property="og:image" content="http://localhost:4000/images/pwa-stats-1280-cropped.png" >
```
Comment (tylersticka) [inline _includes/head.html]: Should the first portion be `{{site.url}}` instead of `http://localhost:4000` hardcoded?

**Comment by derekshirk** [comment]
@tylersticka updated.

**Comment by derekshirk** [comment]
Yay.

**Pull request merged**